add smart completion for ValueMapping

hduelme opened the pull request:
As discussed in #5 I add a smart filter to prevent already mapped source values from being displayed.

@filiphr could you have a look?

Fixes #5

filiphr commented:
Thanks for this PR @hduelme, can we please add some tests for this?

It would also be good not to use `Impl` classes such as `PsiLiteralExpressionImpl`, I think that there is `PsiLiteralExpression`.

hduelme commented:
@filiphr I have replaced the usage of the `Impl` class  by `PsiLiteralExpression`. 
I also added 2 tests;
- one source value already mapped
- all source values already mapped

filiphr commented:
Thanks @hduelme. I've integrated this in https://github.com/mapstruct/mapstruct-idea/commit/2a80757c06ed6e7725dd140f6abe4e6eb058b414
